packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart

```diff
@@ -0,0 +1,332 @@
+/// {@endtemplate}
+class PatchIosCommand extends ShorebirdCommand
+    with
+        ShorebirdConfigMixin,
+        ShorebirdBuildMixin,
+        ShorebirdValidationMixin,
+        ShorebirdArtifactMixin,
+        ShorebirdCodePushClientMixin {
+  /// {@macro patch_ios_command}
+  PatchIosCommand({
+    required super.logger,
+    super.auth,
+    super.buildCodePushClient,
+    super.validators,
+    HashFunction? hashFn,
+    IpaReader? ipaReader,
+  })  : _hashFn = hashFn ?? ((m) => sha256.convert(m).toString()),
+        _ipaReader = ipaReader ?? IpaReader() {
+    argParser
+      ..addOption(
+        'release-version',
+        help: 'The version of the release (e.g. "1.0.0").',
+      )
+      ..addOption(
+        'channel',
+        help: 'The channel the patch should be promoted to (e.g. "stable").',
+        allowed: ['stable'],
+        allowedHelp: {
+          'stable': 'The stable channel which is consumed by production apps.'
+        },
+        defaultsTo: 'stable',
+      )
+      ..addOption(
+        'target',
+        abbr: 't',
+        help: 'The main entrypoint file of the application.',
+      )
+      ..addOption(
+        'flavor',
+        help: 'The product flavor to use when building the app.',
+      )
+      ..addFlag(
+        'force',
+        abbr: 'f',
+        help: 'Patch without confirmation if there are no errors.',
+        negatable: false,
+      )
+      ..addFlag(
+        'dry-run',
+        abbr: 'n',
+        negatable: false,
+        help: 'Validate but do not upload the patch.',
+      );
+  }
+
+  @override
+  String get name => 'ios';
+
+  @override
+  String get description =>
+      'Publish new patches for a specific iOS release to Shorebird.';
+
+  final HashFunction _hashFn;
+  final IpaReader _ipaReader;
+
+  @override
+  Future<int> run() async {
+    try {
+      await validatePreconditions(
+        checkShorebirdInitialized: true,
+        checkUserIsAuthenticated: true,
+        checkValidators: true,
+      );
+    } on PreconditionFailedException catch (error) {
+      return error.exitCode.code;
+    }
+
+    const platform = 'ios';
+    final force = results['force'] == true;
+    final dryRun = results['dry-run'] == true;
+    final flavor = results['flavor'] as String?;
+    final target = results['target'] as String?;
+    final channelName = results['channel'] as String;
+
+    if (force && dryRun) {
+      logger.err('Cannot use both --force and --dry-run.');
+      return ExitCode.usage.code;
+    }
+
+    final shorebirdYaml = getShorebirdYaml()!;
+    final appId = shorebirdYaml.getAppId(flavor: flavor);
+    final App? app;
+    try {
+      app = await getApp(appId: appId, flavor: flavor);
+    } catch (_) {
+      return ExitCode.software.code;
+    }
+
+    if (app == null) {
+      logger.err(
+        '''
+Could not find app with id: "$appId".
+Did you forget to run "shorebird init"?''',
+      );
+      return ExitCode.software.code;
+    }
```

Comment:
This feels like code we can share in a later pass.

Reply:
WIP on https://github.com/shorebirdtech/shorebird/tree/bo/code-push-client-wrapper